WordPress/Classes/Services/PostService.h

```diff
@@ -29,6 +29,13 @@ extern const NSUInteger PostServiceDefaultNumberToSync;
 - (AbstractPost *)findPostWithID:(NSNumber *)postID inBlog:(Blog *)blog;
 
 /**
+ Checks whether a user needs to verify their email before publishing a post.
+
+ @param post
+ */
+- (BOOL)needsVerificationBeforePublishing:(AbstractPost *)post
```

Comment:
I'm not sure whether this really belongs _here_. 

There's also a potential edge case here with this call being synchronous — user could've verified the account, but we wouldn't have refreshed the status. 
On the other hand, making this async would mean having to nest calls to `uploadPost` _inside_ the potential `needsVerification` and that just seems... wrong.

Reply:
(But also: if this were to be async, it feels like it'd belong on `AccountService`. Then every VC that handles posting would need to know about `AccountService` in addition to `PostService`, which also seems... suboptimal.)

Reply:
> I'm not sure whether this really belongs here. [...] it feels like it'd belong on `AccountService`.

If what we really want to check is whether an account's email address has been verified, maybe it would be helpful to think in more generic terms than the specific case of publishing.   In that case, yeah, `AccountService` might be a better fit since it deals with account properties. 

> Then every VC that handles posting would need to know about AccountService in addition to PostService, which also seems... suboptimal.)

True. Would it make sense to factor some of the functionality into a helper class to hide implementation details from the controllers?

Reply:
>If what we really want to check is whether an account's email address has been verified, maybe it would be helpful to think in more generic terms than the specific case of publishing.

There's already a `emailVerified` property on `WPAccount` (and we can reach out to that via `post.blog.account`), but I didn't want to replicate that check (and also to see if it's a WPCom blog or self-hosted one) across three different classes.

If it's fine to not support this anywhere else than Aztec, then I think it doesn't make sense to put it into any `Service` and it can happily live as an implementation detail inside the VC code. WDYT?

Reply:
> then I think it doesn't make sense to put it into any Service and it can happily live as an implementation detail inside the VC code. WDYT?

Makes sense to me :)